### PR TITLE
Notes on backup validity & object locking

### PR DESCRIPTION
**Pull request by kathancox**
Fixes [DOC-3437](https://cockroachlabs.atlassian.net/browse/DOC-3437)

Added language to recommend users enable object locking in their cloud storage buckets to help prevent invalidating backups. Also added a general consideration that messing with backup files could invalidate a backup.

### CODE REVIEW

File: v22.1/backup.md [hunk 1]
```diff
@@ -29,12 +29,13 @@ Because CockroachDB is designed with high fault tolerance, these backups are des
 ## Considerations
 
 - Core users can only take [full backups](take-full-and-incremental-backups.html#full-backups). To use the other backup features, you need an [Enterprise license](enterprise-licensing.html). You can also use [{{ site.data.products.dedicated }}](https://cockroachlabs.cloud/signup?referralId=docs-crdb-backup), which runs [full backups daily and incremental backups hourly](../cockroachcloud/backups-page.html).
-- `BACKUP` is a blocking statement. To run a backup job asynchronously, use the `DETACHED` option. See the [options](#options) below.
+- Interference with backup files in the storage location could invalidate a backup, and therefore, prevent a restore. In v22.1 and later, **we recommend using [object locking](use-cloud-storage-for-bulk-operations.html#object-locking) in your cloud storage bucket.**
```
Comment (livlobo) [inline v22.1/backup.md:32]: "Interference"... feels a little weird? Would "Modifying" work?

File: v22.1/backup.md [hunk 2]
```diff
@@ -45,7 +46,7 @@ To view the contents of an Enterprise backup created with the `BACKUP` statement
...
 - For all other backups, the user must have [read access](security-reference/authorization.html#managing-privileges) on all objects being backed up. Database backups require `CONNECT` privileges, and table backups require `SELECT` privileges. Backups of user-defined schemas, or backups containing user-defined types, require `USAGE` privileges.
 - `BACKUP` requires full read and write permissions to its target destination.
-- **New in v22.1:** `BACKUP` does **not** require delete or overwrite permissions to its target destination. This allows `BACKUP` to write to cloud storage buckets that have [object locking](use-cloud-storage-for-bulk-operations.html#object-locking) configured.
+- **New in v22.1:** `BACKUP` does **not** require delete or overwrite permissions to its target destination. This allows `BACKUP` to write to cloud storage buckets that have [object locking](use-cloud-storage-for-bulk-operations.html#object-locking) configured. We recommend enabling object locking in cloud storage buckets to protect the validity of a backup in the event a restore is necessary.
```
Comment (livlobo) [inline v22.1/backup.md:49]: .. "in the event a restore is necessary" seems redundant?

**Review by livlobo — COMMENTED** [review]
LGTM, minor nits.

**Review by andf-crl — APPROVED** [review]
LGTM with one nit. Thanks!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13943#-:-N4TjUsT1EEGBonsI9wi:b-6xj292)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox and @livlobo)

___
*[`v22.1/backup.md` line 32 at r2](https://reviewable.io/reviews/cockroachdb/docs/13943#-N2vttK2Lqw3I5z17vJV-r2-32:-N4Tiukh89Aug028Q3E7:b-4r5dms) ([raw file](https://github.com/cockroachdb/docs/blob/f068183e7e34ff8e6c772ee4738f957e1603628f/v22.1/backup.md#L32)):*
<details><summary><i>Previously, livlobo wrote…</i></summary><blockquote>

"Interference"... feels a little weird? Would "Modifying" work?
</blockquote></details>

nit: `using` --> `enabling` to match phrasing on target link.
___
*[`v22.1/backup.md` line 38 at r3](https://reviewable.io/reviews/cockroachdb/docs/13943#-N4TiZh_Adq9PGyqb1cb:-N4TiZh_Adq9PGyqb1cc:b-fay257) ([raw file](https://github.com/cockroachdb/docs/blob/e22ed10dccdd233fc3cb66069015beb5119a979f/v22.1/backup.md#L38)):*
> ```Markdown
> - [HTTP storage](use-a-local-file-server-for-bulk-operations.html) is not supported for `BACKUP` and `RESTORE`.
> - Exclude a table's row data from a backup using the [`exclude_data_from_backup`](take-full-and-incremental-backups.html#exclude-a-tables-data-from-backups) parameter.
> - `BACKUP` is a blocking statement. To run a backup job asynchronously, use the `DETACHED` option. See the [options](#options) below.
> ```

Just a tiny callout: this change seems out of line with the originating Jira, PR desc, etc., just in case this change was not intended. If so: ignore me!

**Comment by kathancox** [comment]
> LGTM with one nit. Thanks!
> 
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13943#-:-N4TjUsT1EEGBonsI9wi:b-6xj292)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox and @livlobo)
> 
> _[`v22.1/backup.md` line 32 at r2](https://reviewable.io/reviews/cockroachdb/docs/13943#-N2vttK2Lqw3I5z17vJV-r2-32:-N4Tiukh89Aug028Q3E7:b-4r5dms) ([raw file](https://github.com/cockroachdb/docs/blob/f068183e7e34ff8e6c772ee4738f957e1603628f/v22.1/backup.md#L32)):_
> 
> Previously, livlobo wrote…
> nit: `using` --> `enabling` to match phrasing on target link.
> 
> _[`v22.1/backup.md` line 38 at r3](https://reviewable.io/reviews/cockroachdb/docs/13943#-N4TiZh_Adq9PGyqb1cb:-N4TiZh_Adq9PGyqb1cc:b-fay257) ([raw file](https://github.com/cockroachdb/docs/blob/e22ed10dccdd233fc3cb66069015beb5119a979f/v22.1/backup.md#L38)):_
> 
> > ```gfm
> > - [HTTP storage](use-a-local-file-server-for-bulk-operations.html) is not supported for `BACKUP` and `RESTORE`.
> > - Exclude a table's row data from a backup using the [`exclude_data_from_backup`](take-full-and-incremental-backups.html#exclude-a-tables-data-from-backups) parameter.
> > - `BACKUP` is a blocking statement. To run a backup job asynchronously, use the `DETACHED` option. See the [options](#options) below.
> > ```
> 
> Just a tiny callout: this change seems out of line with the originating Jira, PR desc, etc., just in case this change was not intended. If so: ignore me!

Done! 

And yes, the second item: just a minor adjustment as I was working on the bullets.

**Pull request merged**